src/savegame_json.cpp

```diff
@@ -3023,6 +3024,61 @@ void mission::serialize( JsonOut &json ) const
     json.end_object();
 }
 
+void activity_reason_info::serialize( JsonOut &json ) const
+{
+    json.start_object();
+    json.member( "reason" );
+    json.write_as_string( reason );
+    json.member( "can_do", can_do );
+    if( con_idx ) {
+        json.member( "con_idx", con_idx->id() );
+    } else {
+        json.member( "con_idx", -1 );
```

Comment:
```suggestion
        json.null_member( "con_idx" );
```
For consistency with how `cata::optional` is written.